### ISSUE TITLE
Assertion in `sexp_container_set_special_arg_status`

### ISSUE THREAD

**Issue by Goober5000** [scp-fs2open/fs2open.github.com#5783]
```
Assert: "Sexp_nodes[arg_node].subtype != SEXP_ATOM_CONTAINER_DATA"
File: sexp_container.cpp
Line: 1779
Attempt to use Replace Container Data as special argument option, which isn't supported. Please report!

ntdll.dll! ZwWaitForSingleObject + 20 bytes
KERNELBASE.dll! WaitForSingleObjectEx + 142 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! SCP_DumpStack + 382 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! dump_stacktrace + 61 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! os::dialogs::AssertMessage + 637 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! sexp_container_set_special_arg_status + 224 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! eval_when + 484 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! eval_sexp + 3418 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! mission_process_event + 1148 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! mission_eval_goals + 595 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! game_simulation_frame + 1618 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! game_frame + 631 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! game_do_frame + 183 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! game_do_state + 410 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! gameseq_process_events + 250 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! game_main + 724 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! SDL_main + 359 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! main_getcmdline + 245 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! invoke_main + 50 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! __scrt_common_main_seh + 302 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! __scrt_common_main + 14 bytes
fs2_open_23_3_0_x64_AVX2-DEBUG.exe! WinMainCRTStartup + 14 bytes
KERNEL32.DLL! BaseThreadInitThunk + 29 bytes
ntdll.dll! RtlUserThreadStart + 40 bytes
```

This occurred while loading a Scroll mission in FSO.  Loading it in FRED works fine.

Here is the sexp that tripped the error:
```
$Formula: ( when-argument 
   ( for-container-data 
      &thievery& ( "Get_First" ) 
   )
   ( has-arrived-delay 
      30 
      "Unknown#CDestroyer1" 
   )
   ( add-remove-escort "<argument>" 89 ) 
)
+Name: Add Thieves
+Repeat Count: 1
+Interval: 1
```

According to Cobra, the `"Get_First"` was added automatically by FRED.  It does look like containers can be selected via either "Replace Container Name" or "Replace Container Data" from the dropdown menu.  The latter automatically adds `"Get_First"` while the former does not.

**Comment by jg18** [comment]
I didn't realize before that `for-*` SEXPs also use special argument. Created a PR to fix it. Thanks for reporting this.

> According to Cobra, the `"Get_First"` was added automatically by FRED.

"Replace Container Data" is valid only if it has an attached container modifier, such as `"Get_First"` (for list containers) or a map key (for map containers), since it's for getting data stored _in_ the container. FRED adds a modifier by default, to ensure the SEXP syntax is valid.

> It does look like containers can be selected via either "Replace Container Name" or "Replace Container Data" from the dropdown menu. The latter automatically adds `"Get_First"` while the former does not.

I tried to explain the difference between them in the [FS Wiki](https://wiki.hard-light.net/index.php/SEXP_Container), but I'm sure the page could be improved.

The TL;DR:
- Replace Container Data is for getting data stored in the container.
-  Replace Container Name is for specifying the container as a SEXP argument (e.g., `get-container-size`).

Replace Variable is able to cover both the variable itself as a SEXP argument _and_ the data stored in the variable, because accessing data in variables doesn't require a modifier.

**Comment by Goober5000** [comment]
Gotcha.  I'll go over this with Cobra.

**Comment by Goober5000** [comment]
Closing issue as fixed.
